### ISSUE TITLE
use notify instead

### ISSUE THREAD

**Issue by bbigras** [dermesser/tokio-inotify#2]
I think with https://crates.io/crates/notify it would also work on Windows.

**Comment by dermesser** [comment]
Thanks, that sounds interesting! I'm not 100% sure how to integrate it though; the idea of using mpsc channels is nice but not trivially integrated with tokio (for that, using mio's channel types would be ideal: https://docs.rs/mio/0.6.1/mio/channel/fn.channel.html). Optimally there'd be a type that can be a) registered with the event loop (which would require it implementing `Evented`) and b) subsequently be polled. (b) is probably the easy part here.

I thought I'd seen some mpsc adaptation for tokio, but that might have been some time ago.

But maybe I'm missing some crucial piece from the `notify` crate, in that case: please help me out :-)

**Comment by bbigras** [comment]
> the idea of using mpsc channels is nice but not trivially integrated with tokio

Yeah I realized that after I started playing with the notify crate. I ended up spawning a thread to listen to the channel and sending it to a [futures::sync::mpsc](https://docs.rs/futures/0.1.10/futures/sync/mpsc/index.html) channel. The future channel's receiver can be used like a normal future (like `rx.for_each()`).

I think the solution would be either to rewrite the notify crate to use futures or add windows support to your crate.

**Comment by dermesser** [comment]
Having notify using Futures would obsolete this crate -- probably for the better :-) I have no idea about Windows, unfortunately, so I'm depending on contributors sending patches for that :-(
